### PR TITLE
[Improvement][BatchQuery] Batch query ProcessDefinitions belongs to need failover ProcessInstance.

### PR DESCRIPTION
**Pull request by BongBongBang**
Change query processDefinition one by one to batch during failover.

## Verify this pull request

This pull request is code cleanup without any test coverage.

### CODE REVIEW

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java [hunk 1]
```diff
@@ -78,6 +75,8 @@ ProcessInstance handleCommand(String host,
 
     ProcessDefinition findProcessDefinition(Long processDefinitionCode, int processDefinitionVersion);
 
+    List<ProcessDefinition> findProcessDefinitions(Map<Long, Integer> codeVersionMap);
```
Comment (ruanwenjun) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java:78]: Please don't add new method in `ProcessService` we will not maintain this class, you need to use ProcessDefinitionService of ProcessDefinitionLogService.
```suggestion
    List<ProcessDefinition> findProcessDefinitionsByProcessInstances(List<ProcessInstance> processInstances);
```

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java [hunk 2]
```diff
@@ -462,6 +462,40 @@ public ProcessDefinition findProcessDefinition(Long processDefinitionCode, int v
...
+                        ProcessDefinition processDefinition = processDefineLogMapper.queryByDefinitionCodeAndVersion(entry.getKey(), entry.getValue());
+                        if (processDefinition != null) {
+                            processDefinition.setId(0);
+                        }
+                        return processDefinition;
+                    })
+                    .collect(Collectors.toList());
+            processDefinitions.addAll(complementProcessDefinitions);
+        }
+        return processDefinitions;
```
Comment (ruanwenjun) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java:496]: Can we directly query from `processDefineLogMapper` by using below SQL?
```
select * from t_ds_process_definition_log where (code = xx and version = xx) or (code = xxx and version = xxx)
```

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java (hunk 2 above)
Comment (BongBongBang) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java:496]: Yeah, got it. It's a better idea.

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java (hunk 1 above)
Comment (BongBongBang) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java:78]: I found that there is no `ProcessDefinitionLogService` in module `dolphinscheduler-service`.Do you mean that i need to create a new class in `dolphinscheduler-service` under package `org.apache.dolphinscheduler.service.process`?

Comment (ruanwenjun) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java:78]: Ok, I see the latest code, there is no `ProcessDefinitionLogService`, you can add `findProcessDefinitionsByProcessInstances(List<ProcessInstance> processInstances)` in `ProcessDefinitionDaoImpl` class.

Comment (BongBongBang) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java:78]: But how can i use this method in `MasterFailoverService`? Inject `ProcessDefinitionDaoImpl `class directly ? Does that fit in Dolphin's convention?

Comment (BongBongBang) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java:78]: OK， I see the `DAO` layer assembles the `Mapper` layer, it's kinda suitable for this purpose👌

**Comment by BongBongBang** [comment]
Migrate to #12506

**Comment by caishunfeng** [comment]
> Migrate to #12506

Please keep one valid pr.

**Pull request closed without merging**